app/src/main/java/ch/epfl/favo/view/tabs/favorList/FavorAdapter.java

```diff
@@ -14,28 +14,29 @@
 
 public class FavorAdapter extends ArrayAdapter<Favor> {
 
-    public FavorAdapter(Context context, ArrayList<Favor> favors) {
-        super(context, 0, favors);
-    }
+  public FavorAdapter(Context context, ArrayList<Favor> favors) {
+    super(context, 0, favors);
+  }
 
-    @Override
-    public View getView(int position, View convertView, ViewGroup parent) {
-        // Get the data item for this position
-        Favor favor = getItem(position);
-        assert favor != null;
-        // Check if an existing view is being reused, otherwise inflate the view
-        if (convertView == null) {
-            convertView = LayoutInflater.from(getContext()).inflate(R.layout.favor_list_item, parent, false);
-        }
-        // Lookup view for data population
-        TextView favorTitle = convertView.findViewById(R.id.title);
-        TextView favorDesc = convertView.findViewById(R.id.desc);
-        // Populate the data into the template view using the data object
-        assert favor.getTitle() != null;
-        assert favor.getDescription() != null;
-        favorTitle.setText(favor.getTitle());
-        favorDesc.setText(favor.getDescription());
-        // Return the completed view to render on screen
-        return convertView;
+  @Override
+  public View getView(int position, View convertView, ViewGroup parent) {
+    // Get the data item for this position
+    Favor favor = getItem(position);
+    assert favor != null;
+    // Check if an existing view is being reused, otherwise inflate the view
+    if (convertView == null) {
+      convertView =
+          LayoutInflater.from(getContext()).inflate(R.layout.favor_list_item, parent, false);
     }
+    // Lookup view for data population
+    TextView favorTitle = convertView.findViewById(R.id.title);
+    TextView favorDesc = convertView.findViewById(R.id.desc);
+    // Populate the data into the template view using the data object
```

Comment:
we should be using @Nullable or Try/Catches for in code. Assert should normally only ever be used for testing because in the case that any one of  this is null its going to throw an assertionFailed and the code will not be able to recover/crash

Reply:
it would be best to use a try/catch with defaults in case anything is null because otherwise it will crash

Reply:
Agree, see above for the answer I gave Alex